lib/src/ruby/module.cc

```diff
@@ -953,16 +963,16 @@ namespace facter { namespace ruby {
 
         // Look for "facter" subdirectories on the load path
         for (auto const& directory : ruby.get_load_path()) {
-            // Get the absolute directory name
             boost::system::error_code ec;
-            path dir = absolute(directory);
+            auto dir = canonicalize(directory).string();
```

Comment:
I think we want to keep `dir` as a `path` here, and only make it a `string` down below? (line 979)